### ISSUE TITLE
Thought about Model->duplicate(): Should'nt all fields be dirty?

### ISSUE THREAD

**Issue by PhilippGrashoff** [atk4/data#317]
Hi there,
if I misinterpret the usage of this function, just kill this one.

My Idea about Model->duplicate() function is to provide a copy of an existing record, when saving it ends up as new record in Persistence.
Speaking in terms of UI, the user clicks "Copy", perhaps edits something and then saves the new record.
**As the record did not exist before, IMO all fields having a value should be dirty. At the moment, only the fields that differ from the original record.**

Stumbled across this as my own implementation of Audit relies on dirty fields. Guess atk4\audit does too? Then it will face the same problem with duplicate function.

**Comment by romaninsh** [comment]
duplicate() iirc does not create duplicate of the record. But I agree that the name is miss-leading.

**Comment by PhilippGrashoff** [comment]
Yeah, name sounds as if it could be used to create a copy, which it is not made for reading the code. 
Trying to figure a scenario this function can be used for :)

I think Agile Data could use a "real" copy function where all values are copied AND set to dirty. Something like `$copy = $original->copy();`
